handlers/handler.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package handlers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3metrics/rules"
+)
+
+var (
+	errEmptyNamespace = fmt.Errorf("cannot generate ruleset key with empty namespace")
+)
+
+// HandlerOptions ...
+type HandlerOptions struct {
+	PropagationDelay time.Duration
+	NamespacesKey    string
+	RuleSetKeyFmt    string
+}
+
+// Handler ...
+type Handler struct {
+	store kv.TxnStore
+	opts  HandlerOptions
+}
+
+// NewHandlerOpts ...
+func NewHandlerOpts(propDelay time.Duration, namespacesKey string, rulesetKeyFmt string) HandlerOptions {
+	return HandlerOptions{propDelay, namespacesKey, rulesetKeyFmt}
+}
+
+// NewHandler ...
+func NewHandler(store kv.TxnStore, opts HandlerOptions) Handler {
+	return Handler{store: store, opts: opts}
+}
+
+// RuleSetKey generates a ruleset key for a given namespace
+func (h Handler) RuleSetKey(ns string) string {
+	return fmt.Sprintf(h.opts.RuleSetKeyFmt, ns)
+}
+
+// Persist saves the given ruleset and optional namespace to the backing store.
+// If saveNamespace is false, will only use the namesapces state to ensure that
+// the RuleSet is still valid.
+func (h *Handler) Persist(
```

Comment:
Will change handler to store and have it only worry about persisting and fetching data. Can move the wrappers directly to where they belong.